src/components/PromotedActionsBar.tsx

```diff
@@ -66,8 +70,18 @@ const PromotedActions = {
         icon: Expensicons.Stopwatch,
         text: Localize.translateLocal(`iou.${isTextHold ? 'hold' : 'unhold'}`),
         onSelected: () => {
-            Navigation.dismissModal();
-            ReportUtils.changeMoneyRequestHoldStatus(reportAction);
+            if (!isTextHold) {
+                Navigation.goBack();
+            }
+            const topmostCentralPaneRoute = getTopmostCentralPaneRoute(navigationRef.getRootState() as State<RootStackParamList>);
+
+            if (topmostCentralPaneRoute?.name !== SCREENS.SEARCH.CENTRAL_PANE && isTextHold) {
+                ReportUtils.changeMoneyRequestHoldStatus(reportAction, ROUTES.REPORT_WITH_ID.getRoute(reportAction?.childReportID ?? ''));
```

Comment:
Coming from https://github.com/Expensify/App/issues/45443, pass `backTo` with `childReportID` not highlight the opened report when there is only one expense, and we fixed it here https://github.com/Expensify/App/pull/46268